rpc/eth_api.go

```diff
@@ -431,16 +426,21 @@ func (e *PublicEthAPI) doCall(args CallArgs, blockNr rpc.BlockNumber, globalGasC
 	txEncoder := authutils.GetTxEncoder(ctx.Codec)
 	txBytes, err := txEncoder(tx)
 	if err != nil {
-		return []byte{}, err
+		return sdk.Result{}, err
 	}
 
 	// Transaction simulation through query
 	res, _, err := ctx.QueryWithData("app/simulate", txBytes)
 	if err != nil {
-		return []byte{}, err
+		return sdk.Result{}, err
+	}
+
+	var simResult sdk.Result
+	if err = e.cliCtx.Codec.UnmarshalBinaryLengthPrefixed(res, &simResult); err != nil {
```

Comment:
Definitely a nitpick but this is also an unnecessary error check, but can just leave as is, possibly keeping the the redundant error checks could be better for future proofing if the code changes in the future anyway